### DISCUSSION TITLE
Cava dependencies?

### DISCUSSION THREAD

**Discussion by andrew-weber** [Alexays/Waybar#2198] [Q&A]
I'm a new to Arch & Wayland, however decently experienced w/ Linux/Unix.

When I follow the prompt the readme to install Waybar I can successfully can run the process from the uninstalled version

```
ninja -C build
./build/waybar
[2023-06-01 13:34:42.037] [info] Using configuration file /home/aweb/.config/waybar/config.jsonc
[2023-06-01 13:34:42.037] [info] Using CSS file /home/aweb/.config/waybar/style.css
[2023-06-01 13:34:42.042] [warning] Waybar has been built without rfkill support.
[2023-06-01 13:34:42.050] [warning] Requested height: 30 is less than the minimum height: 37 required by the modules
[2023-06-01 13:34:42.050] [info] Bar configured (width: 1920, height: 37) for output: DP-3
```

However, after running the installation, i get an error for `libcava.so`
```
ninja -C build install
waybar 
waybar: error while loading shared libraries: libcava.so: cannot open shared object file: No such file or directory
```

I've tried installing `cava` via `yay`, but the Waybar source indicates that it requires version `>=.8.4.`, which has never even been released.

All of this was in an attempt to remove this line
```
[2023-06-01 13:34:42.042] [warning] Waybar has been built without rfkill support.
```

I at one point tried building Waybar with the following command:
```
meson build --auto-features=enabled
```

**Comment by LukashonakV** [comment]
Hi @andrew-weber .
Let's clarify which waybar source do you use ? (git or arch aur repo)
`ninja -C build` just tries to build waybar with the assumption : all necessary dependencies are installed in the system.
Original cava project is provided as a standalone application. For the waybar purpose it was created cava fork which is packaging cava application into the library. cava's owner declined PR due to hi is afraid of such big changes. So he suggested to keep libcava in separate fork project. So actually [libcava](https://github.com/LukashonakV/cava) is used by Waybar meson build script.

In order to have waybar built with cava support you should use meson commands
Just use

1.  `make` in waybar git directory
2.  `meson install -C  build`
Meson will automatically download, build, install absent necessary dependencies (like cava)

**Reply by LukashonakV** [reply]
@andrew-weber any feedback?

**Comment by LukashonakV** [comment]
Hi @andrew-weber , the issue with cava was solved. Please check waybar trunk ... or wait for the next Waybar release. Thank you

**Comment by eigenpaul** [comment]
@andrew-weber Did you manage to resolve this? I'm running into the same issue

**Comment by EndorFlem** [comment]
Same thing, nothing can help

**Reply by LukashonakV** [reply]
Which waybar repo do you use?

**Reply by EndorFlem** [reply]
Now build from source works fine, but aur packages are still broken

**Reply by EndorFlem** [reply]
UPD. when you use waybar from build folder (after ninja -C build) it works
When you try to install it (with ninja -C build install), it crashes (libcava.so cannot open shared object file)

**Comment by LukashonakV** [comment]
Hi @andrew-weber , @paulphys , @EndorFlem  can you please check [WIKI](https://github.com/Alexays/Waybar/wiki/Module:-Cava#solving-issues) ? I refreshed the page. Please check **on start Waybar throws an exception "error while loading shared libraries: libcava.so: cannot open shared object file: No such file or directory"** section. It might help you.
Please provide feedback.

**Reply by LukashonakV** [reply]
Hi @paulphys any feedback please  😀

**Reply by LukashonakV** [reply]
Ping @andrew-weber , @paulphys , @EndorFlem

**Reply by eigenpaul** [reply]
I'll give it another try on the weekend

**Reply by LukashonakV** [reply]
Hi @paulphys , have you had a chance to check ?

**Reply by eigenpaul** [reply]
Sorry for the late reply. Following the troubleshoot guide fixes the issue for me. Thanks @LukashonakV
